pkg/operator/controller/machineconfig/mcp.go

```diff
@@ -133,15 +133,15 @@ func (r *MachineConfigReconciler) GetProfilingMCP(name string) *mcv1.MachineConf
 func (r *MachineConfigReconciler) createMCP(ctx context.Context, name string) error {
 	mcp := r.GetProfilingMCP(name)
 
-	if err := ctrlutil.SetControllerReference(r.CtrlConfig, mcp, r.Scheme); err != nil {
-		r.Log.Error(err, "failed to update owner info in MCP", "MCP", name)
+	if err := ctrlutil.SetOwnerReference(r.CtrlConfig, mcp, r.Scheme); err != nil {
```

Comment:
I'm having a doubt about this one. I think this was changed to SetControllerReference very recently

Reply:
Yes, my fault, bad merge conflict resolution. `ControllerRef` must be kept to make the cascade deletion and owner re-queuing working.